src/Core/Web/XMLHttpRequestHelper.js

```diff
@@ -0,0 +1,26 @@
+/**
+ * Generated On: 2015-10-5
+ * Class: XMLHttpRequestHelper. Simple promise-based wrapper around XMLHttpRequest
+ */
+
+
+define('Core/Web/XMLHttpRequestHelper', ['when'], function(when) {
+    return {
+        fetch: function(url, responseType) {
```

Comment:
My _problem_ with this PR is that it adds a level of abstraction that's probably not really needed (standardises on `GET`, with no access to request headers et al.)

Moving to fetch, the calling code would move from:

```js
return XMLHttpRequestHelper.fetch(url, 'arraybuffer').then(function(result) {
    return that.parseXBil(result);
});
```
to
```js
return fetch(url).then(function(response) {
    return that.parseXbil(response.arrayBuffer());
});
```

So, for the 3 cases here, it wouldn't change much things, but is much more flexible.

As for `when` vs. native promises, `when` comes with a polyfill (https://github.com/cujojs/when/blob/master/docs/es6-promise-shim.md), and if that doesn't work, we could use `es6-promise` instead and easily adapt that to a `when` promise AFAICT:
```js
return when.promise(fetch(url)).then(function(response) {
    return that.parseXbil(response.arrayBuffer());
});
```